dydxprotocol/v4-web
# Add historical trading rewards gated by feature flag

no trading rewards:
<img width="1366" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/7370b1d9-0038-4eee-9c39-ca8a44d33118">
with trading rewards:
<img width="1365" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/aafa07f5-04da-49e1-a6d2-2f83d9276a36">

steps to test:
- change network to staging
- trade something (orders must be filled) so you get trading rewards -- you should see a block reward notification
- head to Rewards page, and add `?tradingrewards=1` to end of url
- observe trading rewards summary + historical trading rewards table
- toggle between monthly/weekly/daily - this is more relevant when you have at least 2 days of trading rewards, hence feature flag gated and want to keep testing in staging for 1week+


Shared components changes
- `Output`: component now supports UTC option -- this is because we're showing the time periods in UTC in the historical trading rewards table
- `Table`: enable view more row custom styling

Library changes
- abacus / state manager: adds support for new types + setHistoricalTradingRewardPeriod helper (set the weekly / monthly / daily option) + changes
- `testFlags`: add `tradingreward`
- account state: add `tradingRewards` to account

View changes
- `RewardsPage`: new grid layout
- `RewardsHelpPanel`: localize
- `TradingRewardsSummaryPanel`: just show current week's rewards
- `RewardHistoryPanel` + `TradingRewardHistoryTable`: show list of periodic trading rewards
